services/web/src/screens/Shops/index.js

```diff
@@ -1,121 +1,105 @@
 import React from 'react';
 import { Link } from 'react-router-dom';
-import { Container, Header, Table, Button, Message } from 'semantic-ui-react';
-import { getData } from 'country-list';
-import { formatDateTime } from 'utils/date';
+import { Switch, Route } from 'react-router-dom';
+import { Loader } from 'semantic-ui-react';
+import { Breadcrumbs, NotFound } from 'components';
 import { request } from 'utils/api';
-import { screen } from 'helpers';
 
-import AppWrapper from 'components/AppWrapper';
-import { Layout } from 'components/Layout';
-import { Confirm } from 'components/Semantic';
+import List from './List';
+import Overview from './Overview';
 
-import HelpTip from 'components/HelpTip';
-import EditShop from 'components/modals/EditShop';
-import Filters from 'components/modals/Filters';
-import { SearchProvider } from 'components/data';
+// --- Generator: imports
+import Products from './Products';
+// --- Generator: end
 
-const countries = getData().map(({ code, name }) => ({
-  value: code,
-  text: name,
-  key: code,
-}));
-
-@screen
 export default class Shops extends React.Component {
-  onDataNeeded = async (params) => {
-    return await request({
-      method: 'POST',
-      path: '/1/shops/search',
-      body: params,
-    });
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      shop: null,
+      error: null,
+      onSave: this.fetchShop,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchShop();
+  }
+
+  componentDidUpdate(lastProps) {
+    const { id } = this.props.match.params;
+    if (id !== lastProps.match.params.id) {
+      this.fetchShop();
+    }
+  }
+
+  fetchShop = async () => {
+    const { id } = this.props.match.params;
+    if (id) {
```

Comment:
As per our discussion: The idea is to separate List and Detail views more explicitly:

```
Shops/Detail/index.js
Shops/Detail/Header.js
Shops/Detail/Overview.js
Shops/Detail/Products.js
Shops/List/index.js
```

Reply:
ok updated this ... can you have a look and approve then I'll do the same treatment for Users

Reply:
@andrewplummer I think it's almost there. Just rename `Menu.js` to `Header.js` since it's no longer just a menu, but a header, menu and breadcrumb now

Reply:
Otherwise looks good!

Reply:
tried to change to `Header` but it conflicts with semantic-ui ... let's visit this later